server/util/flagutil/common/common.go

```diff
@@ -38,64 +38,111 @@ func flagTypeFromFlagFuncName(name string) reflect.Type {
 }
 
 type TypeAliased interface {
+	// AliasedType returns the type this flag.Value aliases.
 	AliasedType() reflect.Type
 }
 
 type IsNameAliasing interface {
+	// AliasedName returns the flag name this flag.Value aliases.
 	AliasedName() string
 }
 
+type WrappingValue interface {
+	// WrappedValue returns the value this flag.Value wraps.
+	WrappedValue() flag.Value
+}
+
 type Appendable interface {
+	// AppendSlice appends the passed slice to this flag.Value.
 	AppendSlice(any) error
 }
 
 type DocumentNodeOption interface {
+	// Transform transforms the passed yaml.Node in place.
 	Transform(in any, n *yaml.Node)
+	// Passthrough returns whether this option should be passed to child nodes.
 	Passthrough() bool
 }
 
-// GetTypeForFlag returns the (pointer) Type this flag aliases; this is the same
+type SetValueForFlagNameHooked interface {
+	// SetValueForFlagNameHooked is the hook for flags that is called when the
+	// flag.Value is set by name.
+	SetValueForFlagNameHook()
+}
+
+// GetTypeForFlagValue returns the (pointer) Type this flag aliases; this is the same
 // type returned when defining the flag initially.
-func GetTypeForFlag(flg *flag.Flag) (reflect.Type, error) {
-	if t, ok := flagTypeMap[reflect.TypeOf(flg.Value)]; ok {
+func GetTypeForFlagValue(value flag.Value) (reflect.Type, error) {
+	if v, ok := value.(WrappingValue); ok {
+		return GetTypeForFlagValue(v.WrappedValue())
+	}
+	if t, ok := flagTypeMap[reflect.TypeOf(value)]; ok {
 		return t, nil
-	} else if v, ok := flg.Value.(TypeAliased); ok {
+	} else if v, ok := value.(TypeAliased); ok {
 		return v.AliasedType(), nil
 	}
-	return nil, status.UnimplementedErrorf("Unsupported flag type at %s: %T", flg.Name, flg.Value)
+	return nil, status.UnimplementedErrorf("Unsupported flag type : %T", value)
 }
 
 // SetValueForFlagName sets the value for a flag by name.
-func SetValueForFlagName(name string, i any, setFlags map[string]struct{}, appendSlice bool, strict bool) error {
+func SetValueForFlagName(name string, i any, setFlags map[string]struct{}, appendSlice bool, force bool, strict bool) error {
```

Comment:
nit: document what these bool params do?

Reply:
Documented one and removed the other two, they're not actually necessary anymore.